File: js/baekjoon/step9/bkjn_9506_1.js
const input = require("fs")
  .readFileSync("dev/stdin")
  .toString()
  .trim()
  .split("\n");

for (i = 0; i < input.length - 1; i++) {
  let num = Number(input[i]);
  let numArr = [];
  let index = 1;
  let numAll = 0;

  while (index < num) {
    if (num % index === 0) numArr.push(index);
    index++;
    numAll = numArr.reduce((a, b) => a + b);
  }
  const str = numArr.join(" + ");
  if (num === numAll) {
    console.log(`${num} = ${str}`);
  } else {
    console.log(`${num} is NOT perfect.`);
  }
}

File: python/baekjoon/bkjn_9506_1.py
while True:
  num = int(input())
  num_list = []

  if num == -1:
    break
  
  for i in range(1, num // 2 + 1):
    if num % i == 0:
      num_list.append(i)
  
  if num == sum(num_list):
    print(f"{num} =", " + ".join(map(str, num_list)))
  else:
    print(f"{num} is NOT perfect.")
